drivers/shared/executor/executor_universal_linux.go

```diff
@@ -114,53 +114,73 @@ func (e *UniversalExecutor) statCG(cgroup string) (int, func(), error) {
 	return fd, cleanup, err
 }
 
+// runningFunc is called after task startup and is running.
+//
+// its use case is for moving the executor process out of the task cgroup once
+// the child task process has been started (cgroups v1 only)
+type runningFunc func() error
```

Comment:
Why do you return this functions instead of making them methods of the executor?

Reply:
No strong reason, it's just that the implementions are different between cgroups v1 and v2. Seemed like defining them as closures inside already-version specific functions was a bit cleaner than expanding the number of functions defined on the executor.